Acceptance tests for DNS provider are failing

### Terraform CLI and Provider Versions

N/A

### Terraform Configuration

```terraform
N/A
```


### Expected Behavior

Acceptance tests should be passing

### Actual Behavior

The last successful run of the acceptance tests was 5a49da - https://github.com/hashicorp/terraform-provider-dns/actions/runs/3468260802

The first noted failure was an unrelated dependency bump to goreleaser - https://github.com/hashicorp/terraform-provider-dns/actions/runs/3692207675

All failures are variations of timeouts:
```bash
--- FAIL: TestAccDnsSRVRecordSet_Basic (8.38s)
=== RUN   TestAccDnsTXTRecordSet_Basic
    resource_dns_txt_record_set_test.go:47: Step 1/6 error: Error running apply: exit status 1
        
        Error: Error updating DNS record: read udp 127.0.0.1:43268->127.0.0.1:15353: i/o timeout
        
          with dns_txt_record_set.foo,
          on terraform_plugin_test.tf line 3, in resource "dns_txt_record_set" "foo":
           3:   resource "dns_txt_record_set" "foo" {
```

### Steps to Reproduce
- Running acceptance tests on local machine will produce same timeout:
```bash
$ ./internal/provider/acceptance.sh
```

### How much impact is this issue causing?

Medium

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

--- a/.github/workflows/test.yml
+++ b/.github/workflows/test.yml
@@ -65,7 +65,7 @@ jobs:
       run: echo "127.0.0.1 ns.example.com" | sudo tee -a /etc/hosts
 
     - name: Install krb5-user
-      run: sudo apt-get install krb5-user
+      run: sudo apt-get update && sudo apt-get install krb5-user
 
     - name: Check out code
       uses: actions/checkout@ac593985615ec2ede58e132d2e21d2b1cbd6127c # v3.3.0

--- a/.github/workflows/test.yml
+++ b/.github/workflows/test.yml
@@ -49,7 +49,7 @@ jobs:
       fail-fast: false
       matrix:
         os:
-          - ubuntu-latest
+          - ubuntu-20.04
         terraform:
           - '0.12.*'
           - '0.13.*'

--- a/.github/workflows/test.yml
+++ b/.github/workflows/test.yml
@@ -49,7 +49,7 @@ jobs:
       fail-fast: false
       matrix:
         os:
-          - ubuntu-20.04
+          - ubuntu-latest
         terraform:
           - '0.12.*'
           - '0.13.*'

--- a/internal/provider/acceptance.sh
+++ b/internal/provider/acceptance.sh
@@ -28,8 +28,8 @@ export DNS_UPDATE_PORT=15353
 
 # Run with no authentication
 
-docker run -d --tmpfs /tmp --tmpfs /run \
-	-v /sys/fs/cgroup:/sys/fs/cgroup:ro \
+docker run --privileged --cgroupns=host -d --tmpfs /tmp --tmpfs /run \
+	-v /sys/fs/cgroup:/sys/fs/cgroup:rw \
 	-v /etc/localtime:/etc/localtime:ro \
 	-v $PWD/internal/provider/testdata/named.conf.none:/etc/named.conf:ro \
 	-p 127.0.0.1:15353:53 \
@@ -40,8 +40,8 @@ cleanup_docker
 
 # Run with TSIG authentication (MD5)
 
-docker run -d --tmpfs /tmp --tmpfs /run \
-	-v /sys/fs/cgroup:/sys/fs/cgroup:ro \
+docker run --privileged --cgroupns=host -d --tmpfs /tmp --tmpfs /run \
+	-v /sys/fs/cgroup:/sys/fs/cgroup:rw \
 	-v /etc/localtime:/etc/localtime:ro \
 	-v $PWD/internal/provider/testdata/named.conf.md5:/etc/named.conf:ro \
 	-p 127.0.0.1:15353:53 \
@@ -52,8 +52,8 @@ cleanup_docker
 
 # Run with TSIG authentication (SHA256)
 
-docker run -d --tmpfs /tmp --tmpfs /run \
-	-v /sys/fs/cgroup:/sys/fs/cgroup:ro \
+docker run --privileged --cgroupns=host -d --tmpfs /tmp --tmpfs /run \
+	-v /sys/fs/cgroup:/sys/fs/cgroup:rw \
 	-v /etc/localtime:/etc/localtime:ro \
 	-v $PWD/internal/provider/testdata/named.conf.sha256:/etc/named.conf:ro \
 	-p 127.0.0.1:15353:53 \
@@ -68,16 +68,16 @@ export DNS_UPDATE_SERVER="ns.example.com"
 
 # Run with Kerberos authentication (password authentication)
 
-docker run -d --tmpfs /tmp --tmpfs /run \
-	-v /sys/fs/cgroup:/sys/fs/cgroup:ro \
+docker run --privileged --cgroupns=host -d --tmpfs /tmp --tmpfs /run \
+	-v /sys/fs/cgroup:/sys/fs/cgroup:rw \
 	-v /etc/localtime:/etc/localtime:ro \
 	-p 127.0.0.1:18888:88 \
 	-p 127.0.0.1:18888:88/udp \
 	-p 127.0.0.1:464:464 \
 	-p 127.0.0.1:464:464/udp \
 	--rm --name kdc kdc || failed
-docker run -d --tmpfs /tmp --tmpfs /run \
-	-v /sys/fs/cgroup:/sys/fs/cgroup:ro \
+docker run --privileged --cgroupns=host -d --tmpfs /tmp --tmpfs /run \
+	-v /sys/fs/cgroup:/sys/fs/cgroup:rw \
 	-v /etc/localtime:/etc/localtime:ro \
 	-v $PWD/internal/provider/testdata/named.conf.kerberos:/etc/named.conf:ro \
 	-p 127.0.0.1:15353:53 \
@@ -88,16 +88,16 @@ cleanup_docker
 
 # Run with Kerberos authentication (keytab authentication)
 
-docker run -d --tmpfs /tmp --tmpfs /run \
-	-v /sys/fs/cgroup:/sys/fs/cgroup:ro \
+docker run --privileged --cgroupns=host -d --tmpfs /tmp --tmpfs /run \
+	-v /sys/fs/cgroup:/sys/fs/cgroup:rw \
 	-v /etc/localtime:/etc/localtime:ro \
 	-p 127.0.0.1:18888:88 \
 	-p 127.0.0.1:18888:88/udp \
 	-p 127.0.0.1:464:464 \
 	-p 127.0.0.1:464:464/udp \
 	--rm --name kdc kdc || failed
-docker run -d --tmpfs /tmp --tmpfs /run \
-	-v /sys/fs/cgroup:/sys/fs/cgroup:ro \
+docker run --privileged --cgroupns=host -d --tmpfs /tmp --tmpfs /run \
+	-v /sys/fs/cgroup:/sys/fs/cgroup:rw \
 	-v /etc/localtime:/etc/localtime:ro \
 	-v $PWD/internal/provider/testdata/named.conf.kerberos:/etc/named.conf:ro \
 	-p 127.0.0.1:15353:53 \
@@ -108,16 +108,16 @@ cleanup_docker
 
 # Run with Kerberos authentication (session authentication)
 
-docker run -d --tmpfs /tmp --tmpfs /run \
-	-v /sys/fs/cgroup:/sys/fs/cgroup:ro \
+docker run --privileged --cgroupns=host -d --tmpfs /tmp --tmpfs /run \
+	-v /sys/fs/cgroup:/sys/fs/cgroup:rw \
 	-v /etc/localtime:/etc/localtime:ro \
 	-p 127.0.0.1:18888:88 \
 	-p 127.0.0.1:18888:88/udp \
 	-p 127.0.0.1:464:464 \
 	-p 127.0.0.1:464:464/udp \
 	--rm --name kdc kdc || failed
-docker run -d --tmpfs /tmp --tmpfs /run \
-	-v /sys/fs/cgroup:/sys/fs/cgroup:ro \
+docker run --privileged --cgroupns=host -d --tmpfs /tmp --tmpfs /run \
+	-v /sys/fs/cgroup:/sys/fs/cgroup:rw \
 	-v /etc/localtime:/etc/localtime:ro \
 	-v $PWD/internal/provider/testdata/named.conf.kerberos:/etc/named.conf:ro \
 	-p 127.0.0.1:15353:53 \

--- a/internal/provider/testdata/Dockerfile
+++ b/internal/provider/testdata/Dockerfile
@@ -6,6 +6,11 @@ EXPOSE 464
 
 STOPSIGNAL SIGRTMIN+3
 
+# Systemd backport
+RUN yum -y install  wget
+RUN wget --no-check-certificate https://copr.fedorainfracloud.org/coprs/jsynacek/systemd-backports-for-centos-7/repo/epel-7/jsynacek-systemd-backports-for-centos-7-epel-7.repo -O /etc/yum.repos.d/jsynacek-systemd-centos-7.repo
+RUN yum -y update systemd ; yum clean all
+
 # Install Kerberos client packages and copy in configuration
 RUN yum install -y krb5-workstation && yum update -y && yum clean all
 COPY --chown=root:root krb5.conf /etc/krb5.conf

--- a/internal/provider/testdata/Dockerfile
+++ b/internal/provider/testdata/Dockerfile
@@ -39,6 +39,11 @@ EXPOSE 53
 
 STOPSIGNAL SIGRTMIN+3
 
+# Systemd backport
+RUN yum -y install  wget
+RUN wget --no-check-certificate https://copr.fedorainfracloud.org/coprs/jsynacek/systemd-backports-for-centos-7/repo/epel-7/jsynacek-systemd-backports-for-centos-7-epel-7.repo -O /etc/yum.repos.d/jsynacek-systemd-centos-7.repo
+RUN yum -y update systemd ; yum clean all
+
 # Install Kerberos client packages (should use cached KDC step above)
 RUN yum install -y krb5-workstation && yum update -y && yum clean all
 COPY --chown=root:root krb5.conf /etc/krb5.conf

